packages/jest-cli/src/cli/processArgs.js

```diff
@@ -221,8 +223,26 @@ function processArgs() {
       if (argv.testEnvData) {
         argv.testEnvData = JSON.parse(argv.testEnvData);
       }
+
+      const yargsSpecialOptions = ['$0', '_'];
+      const allowedOptions = Object.keys(options).reduce((acc, option) => {
+        return acc
+          .add(option)
+          .add(options[option].alias);
+      }, new Set(yargsSpecialOptions));
+      const unrecognizedOptions = Object.keys(argv).filter(arg => {
+        return !allowedOptions.has(arg);
+      });
+      if (unrecognizedOptions.length) {
+        throw new Error(
```

Comment:
can we turn this into a `console.warn`? I'd prefer this not to be a breaking change and prevent tests from running on CIs or wherever people use Jest with potentially weird args.